### ISSUE TITLE
[Request]: Update warning message

### ISSUE THREAD

**Issue by tiCkuhAr** [YimMenu/YimMenu#2227]
### Problem

Some people are getting confused why the game crashes for no reason and put themself at risk of getting banned

### Solution

Have the menu Check the GTA build against a static build number. If it does not match, give the user a warning message letting them know. Continued use of the menu could result in a ban or unstable gameplay.

### Reason

So people don't get themselves banned or complain about crashes for no reason.

### Additional context

Obviously, for those of us who know it's best not to use the menu right after a GTA update, it's not a problem, but it's pretty clear that not everyone knows that.

**Comment by Tayipc38** [comment]
People are responsible for potential bans either way, and a disclaimer is already present to those who choose to use the mod. It just takes a little reading to see that there might have been an update. Not saying a build check is a bad idea tho.

**Comment by tiCkuhAr** [comment]
> People are responsible for potential bans either way, and a disclaimer is already present to those who choose to use the mod. It just takes a little reading to see that there might have been an update. Not saying a build check is a bad idea tho.

Yes, there's always a risk of getting yourself banned. That's not what I'm talking about here; I'm talking about people using an outdated version of the menu and getting themselves banned for no reason.

**Comment by szarpii** [comment]
But it should be something like that. When GUI is first time opened after injection - message appears
should be simple in theory, right? something like storage data if menu was opened, if 0 = show, else, do the stuff normall
Also, message where people are invited to being good modders, instead of toxic trash, to be those who kick out bad ones, and if they have to abuse their powers, just wait for some toxic player, toxic modder or toxic griefer, not hard to find.. just .. people please dont be trash.. i described more ideas in one of bug reports but it closed bruh

**Comment by Tayipc38** [comment]
None of you read what I wrote.

**Comment by tiCkuhAr** [comment]
> None of you read what I wrote.

I read your entire message. It just did not make any sense. It sounded like you were just talking about how there's always a risk of being banned from using the menu in general, and that has nothing to do with this request also I just tried loading an old version of the menu, and it did not tell me anything about it being outdated or incompatible with the current version of GTA, so if there's already something like that in the menu, I don't think it's working.

**Comment by Tayipc38** [comment]
> > None of you read what I wrote.
> 
> I read your entire message. It just did not make any sense. It sounded like you were just talking about how there's always a risk of being banned from using the menu in general, and that has nothing to do with this request also I just tried loading an old version of the menu, and it did not tell me anything about it being outdated or incompatible with the current version of GTA, so if there's already something like that in the menu, I don't think it's working.

I have also implied that the idea presented wasn't entirely unreasonable. What did not make sense is your response.

**Comment by eggrolls-repu** [comment]
Let people do what they want. I was going to type more but simple is bettr

**Comment by eggrolls-repu** [comment]
> None of you read what I wrote.

Because we don’t care😭

**Comment by ghost** [comment]
At first I thought what you typed was some kind of copypasta tbh but I actually read what you typed and yeah a build number check makes sense but also hm is this possible? Like, if it’s injected and it shows the warning and say the menu is detected, if you just close the warning or I guess say to abort would you still get banned? I know that kiddions has a warning if the build doesn’t match but that’s different since it’s an external. Anyway I typed too much in bad at typing im tired gn

**Comment by tiCkuhAr** [comment]
> > > None of you read what I wrote.
> > 
> > 
> > I read your entire message. It just did not make any sense. It sounded like you were just talking about how there's always a risk of being banned from using the menu in general, and that has nothing to do with this request also I just tried loading an old version of the menu, and it did not tell me anything about it being outdated or incompatible with the current version of GTA, so if there's already something like that in the menu, I don't think it's working.
> 
> I have also implied that the idea presented wasn't entirely unreasonable. What did not make sense is your response.

That end part did make sense. "Not saying a build check is a bad idea, though." It was the rest it that had me confused.

**Comment by lonelybud** [comment]
I think we should add check for build number.

**Comment by ghost** [comment]
Yes

**Comment by lonelybud** [comment]
i also see that people don't use latest build. Since this open source and regularly update they should keep track of updates.

**Comment by tiCkuhAr** [comment]
> i also see that people don't use latest build. Since this open source and regularly update they should keep track of updates.

A hash check that compares the user's current menu hash to the last release hash would be better for dealing with that.

**Comment by szarpii** [comment]
Yup, and if new version, THERE IS NEW VERSION AVAILABLE! DO 
YOU WANT TO UPDATE? I can help with auto updating feature.
